### ISSUE TITLE
Different behavior regarding strings in data.frame between R 3.6.3 and R 4.0.3

### ISSUE THREAD

**Issue by jbabyhacker** [extendr/extendr#49]
I am seeing strange behavior with data.frame between R 3.6.3 and R 4.0.3. I have reproduced this on both Ubuntu 20.04.1 and Mac OS 11.0.1. I am using extendr version 0.1.10.

In R 3.6.3 the following code prints out: [[1, 2, 3], [3.0, 4.0, 5.0], [6, 7, 8]]
In R 4.0.3 the following code prints out: [["abcd", "def", "fg"], [3.0, 4.0, 5.0], [6, 7, 8]]

I have spent a few hours at this. Any suggestions on where to look to debug this would be most grateful. 

```Rust
fn main() {
    start_r();
    let x_vec = &["abcd", "def", "fg"][..];
    let y_vec = &[3.0, 4.0, 5.0][..];
    let z_vec = &[6, 7, 8][..];

    let data_frame = data_frame!(x = x_vec.clone(), y = y_vec.clone(), z = z_vec.clone());

    println!("{:?}", data_frame);

    let c = data_frame.list_iter().unwrap().collect_vec();
    if let Some(v) = c[0].str_iter() {
        // Evaluated in R 4.0.3
        // v.collect_vec() is ["abcd", "def", "fg"]
        assert_eq!(v.collect_vec(), x_vec);
    } else {
        // c[0] evaluates to [1,2,3] and is of type i32
        // str_iter() fails in R 3.6.3
    }
    assert_eq!(c[1].as_f64_slice(), Some(y_vec));
    assert_eq!(c[2].as_i32_slice(), Some(z_vec));
    end_r();
}
```

**Comment by clauswilke** [comment]
Are you using a libR-sys compiled for the specific version of R?

**Comment by jbabyhacker** [comment]
Only one version of R was installed at a time. So it is just going to used whatever version is currently installed. I had version 3.6.3 installed, ran the test. Uninstalled it. Then installed 4.0.3, and ran the test.

**Comment by clauswilke** [comment]
It's possible something isn't right in the FFI for R 3.6. That would be a question for @andy-thomason.

Did you run `cargo test`? Did it succeed with both R versions?

**Comment by jbabyhacker** [comment]
The tests all pass. I have found the issue. The [blog post](https://developer.r-project.org/Blog/public/2020/02/16/stringsasfactors/index.html) describes the change between strings in data.frame between pre R 4.x and post R 4.x releases. By updating the code to set stringsAsFactors to false, the data.frame now works in pre 4.x releases.

```rust
let data_frame = data_frame!(x = x_vec.clone(), y = y_vec.clone(), z = z_vec.clone(), stringsAsFactors = false);
```

**Comment by clauswilke** [comment]
Oh, I see, it's the data frame `stringsAsFactors` issue. But this suggests that `str_iter()` doesn't handle factors correctly yet. I would expect it to be able to iterate over a factor and return the correct labels, not the numeric values, for each level.

**Comment by andy-thomason** [comment]
I'll take a look at `str_iter()` to see if we can handle factors.

It is good to see someone using the R macros.

**Comment by andy-thomason** [comment]
I've popped in a fix to enable str_iter() for factors.

This also adds a few more helper functions to Robj.

**Comment by clauswilke** [comment]
This should now be fixed thanks to #58. If that's not the case, please comment here with any remaining issues and I'll reopen.
